main.rb

```diff
@@ -0,0 +1,229 @@
+        return true if yield i
+      end
+    elsif arg[0]
+      if arg[0] == Regexp
+        my_each do |i|
+          return true if arg[0].match?(i)
+        end
+      elsif arg[0].is_a?(Class)
+        my_each do |i|
+          return true if i.is_a?(arg[0])
+        end
+      else
+        my_each do |i|
+          return true if i == arg[0]
+        end
+      end
+    else
+      my_each do |i|
+        return true if i
+      end
+    end
+    counter
+  end
+
+  def my_none?(*arg)
+    counter = true
+    if block_given?
+      my_each do |i|
+        counter = false if yield i
+      end
+    elsif arg[0]
+      if arg[0] == Regexp
+        my_each do |i|
+          counter = false if arg[0].match?(i)
+        end
+      elsif arg[0].is_a?(Class)
+        my_each do |i|
+          counter = false if i.is_a?(arg[0])
+        end
+      else
+        my_each do |i|
+          counter = false if i == arg[0]
+        end
+      end
+    else
+      my_each do |i|
+        counter = false if i
+      end
+    end
+    counter
+  end
+
+  def my_count(*arg)
+    counter = 0
+    if block_given?
+      my_each do |i|
+        counter += 1 if yield i
+      end
+    elsif arg[0]
+      if arg[0] == Regexp
+        my_each do |i|
+          counter += 1 if arg[0].match?(i)
+        end
+      elsif arg[0].is_a?(Class)
+        my_each do |i|
+          counter += 1 if i.is_a?(arg[0])
+        end
+      else
+        my_each do |i|
+          counter += 1 if i == arg[0]
+        end
+      end
+    else
+      my_each do |i|
+        counter += 1 if i
+      end
+    end
+    counter
+  end
+
+  def my_map(*arg)
+    map_val = []
+    if block_given? && arg[0].is_a?(Proc)
+      my_each do |i|
+        map_val << i if arg[0].call(i)
+      end
+    elsif block_given?
+      my_each do |i|
+        map_val << i if yield i
+      end
+    elsif arg[0]
+      if arg[0] == Regexp
+        my_each do |i|
+          map_val << i if arg[0].match?(i)
+        end
+      elsif arg[0].is_a?(Class)
+        my_each do |i|
+          map_val << i if i.is_a?(arg[0])
+        end
+      elsif arg[0].is_a?(Proc)
+        my_each do |i|
+          map_val << i if arg[0].call(i)
+        end
+      else
+        my_each do |i|
+          map_val << i if i == arg[0]
+        end
+      end
+    else
+      my_each do |i|
+        map_val << i if i
+      end
+    end
+    map_val
+  end
+
+  def my_inject(*arg)
+    if block_given? && arg[0]
+      result = arg[0]
+      my_each do |i|
+        result = yield result, i
+      end
+    elsif block_given?
+      result = self[0]
+      my_each do |i|
+        result = yield result, i
+      end
+    elsif arg[0] && arg[1]
+      result = arg[0]
+      for i in (0..to_a.length - 1) do
+        result = result.send(arg[1], to_a[i])
+      end
+    else
+      result = to_a[0]
+      for i in (1..to_a.length - 1) do
+        result = result.send(arg[0], to_a[i])
+      end
+    end
+    result
+  end
+end
```

Comment:
# my_inject 
- [ ] `my_inject` raises a "LocalJumpError" when no block or argument is given

      Failure/Error: expect { array.my_inject }.to raise_error(LocalJumpError)

        expected LocalJumpError, got #<TypeError: nil is not a symbol nor a string> with backtrace:
##

- [ ] `my_inject` when a block is given without an initial value combines all elements of enum by applying a binary operation, specified by a block

      Failure/Error: expect(array.my_inject(&operation)).to eq array.inject(&operation)

        expected: 345
             got: 351

##

- [ ] `my_inject` when a block is given without an initial value combines all elements of enum by applying a binary operation, specified by a block::range

      Failure/Error: actual = range.my_inject { |prod, n| prod * n }

      NoMethodError:
        undefined method `[]' for 5..50:Range